lib/scholar/metrics/classification.ex

```diff
@@ -586,86 +589,179 @@ defmodule Scholar.Metrics.Classification do
   end
 
   @doc """
-  Calculates F1 score given rank-1 tensors which represent
+  Calculates F-beta score given rank-1 tensors which represent
   the expected (`y_true`) and predicted (`y_pred`) classes.
 
   If all examples are true negatives, then the result is 0 to
   avoid zero division.
 
   ## Options
 
-  #{NimbleOptions.docs(@f1_score_schema)}
+  #{NimbleOptions.docs(@fbeta_score_schema)}
 
   ## Examples
 
       iex> y_true = Nx.tensor([0, 1, 1, 1, 1, 0, 2, 1, 0, 1], type: :u32)
       iex> y_pred = Nx.tensor([0, 2, 1, 1, 2, 2, 2, 0, 0, 1], type: :u32)
-      iex> Scholar.Metrics.Classification.f1_score(y_true, y_pred, num_classes: 3)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.tensor(1), num_classes: 3)
```

Comment:
```suggestion
      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.u32(1), num_classes: 3)
```

Reply:
I wasn't aware of these "short-hand functions", I think it makes sense. Just a question, any reason to not use the smaller versions, u8 and f16?

Reply:
@0urobor0s getting a bit more into the gist of your question, it wouldn't make much of a difference because the inferred type for the metric would be f32 because it comes from integer typed tensors.

I believe that by using y_true and y_pred as f16, then it would make a difference if beta was f32/f64 or not. Those two would upcast everything to that same type. You can explore a bit on this by looking at `Nx.Type.merge`

Reply:
Makes sense, I'll take a look at those. Thank you all.